src/solver/solverUtils.F90

```diff
@@ -392,6 +392,272 @@ subroutine gridVelocitiesFineLevel(useOldCoor, t, sps)
+    ke_l = flowDoms(nn, 1, sps)%ke  
+
+    do k=1, ke_l
+       do j=1, je_l
+          do i=1, ie_l
+
+             s(i, j, k, 1) = velxGrid
+             s(i, j, k, 2) = velyGrid             
+             s(i, j, k, 3) = velzGrid
+
+          end do
+       end do
+    end do
+
+    ! the velocity contributed from mesh deformation 
+    do mm=1, nTimeIntervalsSpectral
+
+       ie_l = flowDoms(nn, 1, mm)%ie
+       je_l = flowDoms(nn, 1, mm)%je
+       ke_l = flowDoms(nn, 1, mm)%ke
+       
+       do k=1,ke_l
+          do j=1,je_l
+             do i=1,ie_l
+
+                x_vc = eighth*(flowDoms(nn, 1, mm)%x(i-1,j-1,k-1,1) + flowDoms(nn, 1, mm)%x(i,j-1,k-1,1) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k-1,1) + flowDoms(nn, 1, mm)%x(i,j,  k-1,1) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j-1,k,  1) + flowDoms(nn, 1, mm)%x(i,j-1,k,  1) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k,  1) + flowDoms(nn, 1, mm)%x(i,j,  k,  1))
+
+                y_vc = eighth*(flowDoms(nn, 1, mm)%x(i-1,j-1,k-1,2) + flowDoms(nn, 1, mm)%x(i,j-1,k-1,2) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k-1,2) + flowDoms(nn, 1, mm)%x(i,j,  k-1,2) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j-1,k,  2) + flowDoms(nn, 1, mm)%x(i,j-1,k,  2) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k,  2) + flowDoms(nn, 1, mm)%x(i,j,  k,  2))
+
+	              z_vc = eighth*(flowDoms(nn, 1, mm)%x(i-1,j-1,k-1,3) + flowDoms(nn, 1, mm)%x(i,j-1,k-1,3) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k-1,3) + flowDoms(nn, 1, mm)%x(i,j,  k-1,3) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j-1,k,  3) + flowDoms(nn, 1, mm)%x(i,j-1,k,  3) &
+                     +         flowDoms(nn, 1, mm)%x(i-1,j,  k,  3) + flowDoms(nn, 1, mm)%x(i,j,  k,  3))
+
+
+                s(i,j,k,1) = s(i,j,k,1) + dscalar(1, sps, mm)*x_vc
+                s(i,j,k,2) = s(i,j,k,2) + dscalar(1, sps, mm)*y_vc
+                s(i,j,k,3) = s(i,j,k,3) + dscalar(1, sps, mm)*z_vc
+
+             end do
+          end do
+       end do
+
+    end do
+     
+!
+!            ************************************************************
+!            *                                                          *
+!            * Normal grid velocities of the faces.                     *
+!            *                                                          *
+!            ************************************************************
+!
+    ! sFaceI=	dot(sI, v)=dot(sI, v_freestream + v_grid)=dot(sI, v_freestream) + dot(sI, v_grid)
+    ! sFaceJ, sFaceK same rule!
+
+    ! dot(sI, v_freestream)
+    ie_l = flowDoms(nn, 1, sps)%ie
+    je_l = flowDoms(nn, 1, sps)%je
+    ke_l = flowDoms(nn, 1, sps)%ke  
+
+
+    ! i
+    do k=1, ke_l
+       do j=1, je_l
+          do i=0, ie_l
+
+             sFaceI(i, j, k) = velxGrid*sI(i, j, k, 1) + velyGrid*sI(i, j, k, 2) &
```

Comment:
Is `velGrid` what the comments refer to as `v_freestream`? This is pretty confusing because there is also a `v_grid` in the comments.

Reply:
That is right. Modified.